llvm/lib/Target/AArch64/AArch64SLSHardening.cpp

```diff
@@ -183,15 +183,13 @@ static const struct ThunkNameAndReg {
 namespace {
 struct SLSBLRThunkInserter : ThunkInserter<SLSBLRThunkInserter> {
   const char *getThunkPrefix() { return SLSBLRNamePrefix; }
-  bool mayUseThunk(const MachineFunction &MF, bool InsertedThunks) {
-    if (InsertedThunks)
-      return false;
+  bool mayUseThunk(const MachineFunction &MF) {
+    // FIXME: ComdatThunks is only accumulated until the first thunk is created.
```

Comment:
I don't fully understand this FIXME very well. Could you explain this in a bit more detail?

Reply:
Not sure how to explain it in a single line - using `&=` seemed a bit confusing to me at first, as it suggests that `ComdatThunks` is accumulated and at the end is used to determine if the thunk functions should be marked comdat. And it really is, but due to the usage of MachineFunctionPass, it is only accumulated until the first MF that triggers thunk insertion.

Reply:
Right, with that longer explanation here, I think I understand.
I'd say that it is fine to use more than one line to explain a FIXME.

IIUC, this means that if a comdat thunk was emitted because one function didn't request a no-comdat; and later another function requires a "no-comdat" thunk to be emitted, this can not be done.

I don't fully remember why we had to introduce the "NoComdat" functionality (but have just looked up that it was introduced as part of https://reviews.llvm.org/D100546). 

IIRC, based on the rationale given on https://reviews.llvm.org/D100546, the no-comdat functionality was needed for projects where the target environment does not support comdat functions. That suggests that for such projects, all functions should be marked to no-comdat.

Maybe it makes more sense to assert on when different nocomdat settings on different MachineFunctions would be observed rather than silently producing incorrect code (a comdat thunk, when a no-comdat thunk was requested?)

Reply:
> IIRC, based on the rationale given on https://reviews.llvm.org/D100546, the no-comdat functionality was needed for projects where the target environment does not support comdat functions.

IIUC comdat thunks are not supported on MachO targets, for example.

> Maybe it makes more sense to assert on when different nocomdat settings on different MachineFunctions would be observed rather than silently producing incorrect code

Sounds reasonable, updated the PR, thanks.